ui/src/app/intro/views/MaasIntro/MaasIntro.tsx

```diff
@@ -64,61 +68,89 @@ const MaasIntro = (): JSX.Element => {
-                maas_name: values.name,
-                upstream_dns: values.upstreamDns,
-              })
-            );
-            if (mainArchive && mainArchive.url !== values.mainArchiveUrl) {
+        <>
+          <FormikForm<MaasIntroValues>
+            allowUnchanged
+            buttonsBordered={false}
+            cleanup={configActions.cleanup}
+            editable={!showSkip}
+            errors={errors}
+            initialValues={{
+              httpProxy: httpProxy || "",
+              mainArchiveUrl: mainArchive?.url || "",
+              name: maasName || "",
+              portsArchiveUrl: portsArchive?.url || "",
+              upstreamDns: upstreamDns || "",
+            }}
+            onSaveAnalytics={{
+              action: "Saved",
+              category: "Intro",
+              label: "Intro form",
+            }}
+            onSubmit={(values) => {
+              dispatch(configActions.cleanup());
+              dispatch(repoActions.cleanup());
               dispatch(
-                repoActions.update({
-                  id: mainArchive.id,
-                  name: mainArchive.name,
-                  url: values.mainArchiveUrl,
+                configActions.update({
+                  http_proxy: values.httpProxy,
+                  maas_name: values.name,
+                  upstream_dns: values.upstreamDns,
                 })
               );
-            }
-            if (portsArchive && portsArchive.url !== values.portsArchiveUrl) {
-              dispatch(
-                repoActions.update({
-                  id: portsArchive.id,
-                  name: portsArchive.name,
-                  url: values.portsArchiveUrl,
-                })
-              );
-            }
-          }}
-          saving={saving}
-          saved={saved}
-          savedRedirect={introURLs.images}
-          submitLabel="Save and continue"
-          validationSchema={MaasIntroSchema}
-        >
-          <NameCard />
-          <ConnectivityCard />
-        </FormikForm>
+              if (mainArchive && mainArchive.url !== values.mainArchiveUrl) {
+                dispatch(
+                  repoActions.update({
+                    id: mainArchive.id,
+                    name: mainArchive.name,
+                    url: values.mainArchiveUrl,
+                  })
+                );
+              }
+              if (portsArchive && portsArchive.url !== values.portsArchiveUrl) {
+                dispatch(
+                  repoActions.update({
+                    id: portsArchive.id,
+                    name: portsArchive.name,
+                    url: values.portsArchiveUrl,
+                  })
+                );
+              }
+            }}
+            saving={saving}
+            saved={saved}
+            savedRedirect={introURLs.images}
+            secondarySubmit={() => {
+              setShowSkip(true);
+            }}
+            secondarySubmitLabel="Skip"
```

Comment:
I wonder if this should say "Skip setup" just to make it a bit clearer that you're not just skipping the first screen.